### PR TITLE
Issue 9 implement class StudyCourseState

### PR DESCRIPTION
**Pull request by krzhld**
Added interface TimerObserver and abstract class StudyCourseState

- TimerObserver includes itself two empty functions
- Abstract class StudyCourseState inherites from the interface. There are val integer property "price", empty function buyNextCourse, which can change object "student" of class "Player". Functions from interface were overrided, but they are still empty

Closes: #9

### CODE REVIEW

**Review by Bazalii — APPROVED** [review]
Everything is fine too. Sorry, that I wrote yesterday that you would have to write realisation for functions in _abstract_ class...

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package android.clicker.school_live_simulator
+
+abstract class StudyCourseState: TimerObserver {
+    abstract val price: Int
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:4]: Price field should have protected access type.

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    abstract fun buyNextCourse(var student: Player) {
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:5]: Where documentation commentaries about what this method should do?

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]
Documentation commentaries should be set in abstract structures as classes and interfaces, then it can be only linked on parent's doc.

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt (hunk 1 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt (hunk 2 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:19]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt [hunk 3]
```diff
@@ -1,6 +1,14 @@
 package android.clicker.school_live_simulator
 
+/**
+ * @author      Kirill Zheludev kirill.zheludev@yandex.ru
+ * @version     1.0                 (current version number of program)
+ * @since       1.0          (the version of the package this class was first added to)
+ */
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt:7]: Please, delete these lines.

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt [hunk 4]
```diff
@@ -1,13 +1,27 @@
 package android.clicker.school_live_simulator
 
+/**
+ * @author      Kirill Zheludev kirill.zheludev@yandex.ru
+ * @version     1.0                 (current version number of program)
+ * @since       1.0          (the version of the package this class was first added to)
+ */
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:7]: Please, delete these lines.

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt [hunk 5]
```diff
@@ -1,6 +1,14 @@
...
+
 interface TimerObserver {
     fun timerTickHandler()
+	
     fun timerEndHandler()
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt]: Where is documentation comments?

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt (hunk 3 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt (hunk 4 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt (hunk 5 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt]: Fixed, but I'm not sure about purposes of functions, so comments may be little incorrect

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package android.clicker.school_live_simulator
+
+abstract class StudyCourseState: TimerObserver {
+    /**
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:4]: Please, use same style everywhere.

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package android.clicker.school_live_simulator
+
+interface TimerObserver {
+	/**
+	 * Function which response 
+	 * for timer tick 
+	*/
+    fun timerTickHandler()
```
[the remark this answers is not in the document]
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt:8]: I should only add curly brackets at functions?

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt (hunk 6 above)
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:4]: Should I only add curly brackets at functions?

Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/StudyCourseState.kt:4]: Done

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt]: Changed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+	
+	/**
+	 * Function which end timer
+	*/
+    fun timerEndHandler()
```
[the remark this answers is not in the document]
Comment (krzhld) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/TimerObserver.kt]: Changed

**Review by Ferrociraptus — APPROVED** [review]

**Review by Bazalii — APPROVED** [review]

**Pull request merged**